guides/source/upgrading_ruby_on_rails.md

```diff
@@ -276,6 +276,22 @@ class Notifier < ActionMailer::Base
 end
 ```
 
+### Foreign Key support
+
+Rails 4.2 adds basic support for foreign keys. If the
+[Foreigner gem](https://github.com/matthuhiggins/foreigner) were been used to
+get support for foreign keys in the app, you can consider removing the gem
+from Gemfile on upgrading to 4.2 and using the default support.
+
+But remember that some optional and adapter specific options might not work as
+expected, since Rails has just basic support. The naming convention for
+generating the constraint names is also different (the names are randomly
+generated), but this shouldn't cause problems.
```

Comment:
the names are no longer randomly generated. They are still not human readable but are consistent across migration runs.